File: uri/strings/array_hash/1257.cpp
#include <iostream>
#include <string>

using namespace std;

int main() {
    int n;
    cin >> n;
    
    for (int t = 0; t < n; t++) {
        int l;
        cin >> l;
        cin.ignore();
        
        int hash = 0;
        
        for (int i = 0; i < l; i++) {
            string line;
            getline(cin, line);
            
            for (int j = 0; j < line.length(); j++) {
                int value = (line[j] - 'A') + i + j;
                hash += value;
            }
        }
        
        cout << hash << endl;
    }
    
    return 0;
}

File: uri/strings/array_hash/1257.rs
use std::io::{self, BufRead};

fn main() {
    let stdin = io::stdin();
    let mut lines = stdin.lock().lines();
    
    let n: usize = lines.next().unwrap().unwrap().trim().parse().unwrap();
    
    for _ in 0..n {
        let l: usize = lines.next().unwrap().unwrap().trim().parse().unwrap();
        
        let mut hash = 0;
        
        for i in 0..l {
            if let Some(Ok(line)) = lines.next() {
                for (j, ch) in line.chars().enumerate() {
                    let value = (ch as i32 - 'A' as i32) + i as i32 + j as i32;
                    hash += value;
                }
            }
        }
        
        println!("{}", hash);
    }
}
